Sources/RInAppMessaging/Views/CarouselView.swift

```diff
@@ -0,0 +1,134 @@
+        layoutIfNeeded()
+    }
+
+    private func setupPageControl() {
+        carouselPageControl.numberOfPages = images.count
+        carouselPageControl.currentPage = 0
+        carouselPageControl.currentPageIndicatorTintColor = .systemBlue
+        carouselPageControl.pageIndicatorTintColor = .lightGray
+        carouselPageControl.addTarget(self, action: #selector(pageControlValueChanged), for: .valueChanged)
+    }
+
+    private func setupCollectionView() {
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.showsHorizontalScrollIndicator = false
+        collectionView.isPagingEnabled = true
+
+        if let layout = collectionView.collectionViewLayout as? UICollectionViewFlowLayout {
+            layout.scrollDirection = .horizontal
+            layout.minimumLineSpacing = 0
+        }
+        collectionView.register(CarouselCell.self, forCellWithReuseIdentifier: "CarouselCell")
+    }
+
+    func setPageControlVisibility(isHdden: Bool) {
+        carouselPageControl.isHidden = isHdden
+    }
+
+    @objc private func pageControlValueChanged() {
+        let currentPage = carouselPageControl.currentPage
+        collectionView.scrollToItem(at: IndexPath(item: currentPage, section: 0), at: .centeredHorizontally, animated: true)
+    }
+}
+
+extension CarouselView: UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return images.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: CarouselCell.identifier, for: indexPath) as? CarouselCell else {
+            return UICollectionViewCell()
+        }
+
+        cell.configure(with: images[indexPath.item],
+                       altText: altTexts[indexPath.item] ?? "carousel_image_load_error".localized)
+        return cell
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let itemWidth = collectionView.frame.width
+        let itemHeight = itemWidth * getMaxImageAspectRatio()
+
+        return  CGSize(width: itemWidth, height: itemHeight)
+    }
+    
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        let page = Int(scrollView.contentOffset.x / collectionView.frame.width)
+        carouselPageControl?.currentPage = page
+    }
+
+    func getMaxImageAspectRatio()-> CGFloat {
+        guard let maxImage = images.compactMap({ $0 }).max(by: { $0.size.height < $1.size.height }) else {
+            return .zero
+        }
+        return maxImage.size.height / maxImage.size.width
+    }
+
+    func getCarouselData(from carousel: Carousel?) {
```

Comment:
this can be moved from here as it's not part of collection view data source/delegate. 
Function name can be renamed as it's not returning any value to justify the `getCarouselData`